packages/experience/src/apis/utils.ts

```diff
@@ -1,26 +1,51 @@
-import { InteractionEvent } from '@logto/schemas';
+import { InteractionEvent, type VerificationCodeIdentifier } from '@logto/schemas';
 
-import { UserFlow } from '@/types';
+import { type ContinueFlowInteractionEvent, UserFlow } from '@/types';
 
-import type { SendVerificationCodePayload } from './interaction';
-import { putInteraction, sendVerificationCode } from './interaction';
+import { initInteraction, sendVerificationCode } from './experience';
 
 /** Move to API */
 export const sendVerificationCodeApi = async (
   type: UserFlow,
-  payload: SendVerificationCodePayload
+  identifier: VerificationCodeIdentifier,
+  interactionEvent?: ContinueFlowInteractionEvent
 ) => {
-  if (type === UserFlow.ForgotPassword) {
-    await putInteraction(InteractionEvent.ForgotPassword);
-  }
-
-  if (type === UserFlow.SignIn) {
-    await putInteraction(InteractionEvent.SignIn);
+  switch (type) {
+    case UserFlow.SignIn: {
+      await initInteraction(InteractionEvent.SignIn);
+      return sendVerificationCode(InteractionEvent.SignIn, identifier);
+    }
+    case UserFlow.Register: {
+      await initInteraction(InteractionEvent.Register);
+      return sendVerificationCode(InteractionEvent.Register, identifier);
+    }
+    case UserFlow.ForgotPassword: {
+      await initInteraction(InteractionEvent.ForgotPassword);
+      return sendVerificationCode(InteractionEvent.ForgotPassword, identifier);
+    }
```

Comment:
can we merge them into one? e.g. mapping `UserFlow` to `InteractionEvent`

Reply:
That is a legacy issue, we had a `UserFlow.Continue`.  `UserFlow` can not be strictly mapped to `InteractionEvent`.  

I'll create an issue for this. Needs some refactor work.

Reply:
For now, create a map for it